### ISSUE TITLE
STS 3.9.1: RequestMapping View shows all the Resource URL with the same / value 

### ISSUE THREAD

**Issue by manueljordan** [spring-projects/spring-ide#227]
Hello STS developers

I have read:

* [request mappings view doesn't respect @GetMapping, @PostMapping etc. #101](https://github.com/spring-projects/spring-ide/issues/101)

Now I am working with:

```
Spring Tool Suite 

Version: 3.9.1.RELEASE
Build Id: 201710111105
Platform: Eclipse Oxygen.1a (4.7.1a)
```
Thus for a `Spring Framework` Web Application working with `Gradle` through _multi-modules_, for a module named `web-29-controller`, right click -> `Spring Tools` -> `Add Spring Project Nature`

For that module I have:

```
@Configuration
@RequestMappingsViewProfile
@ComponentScan(basePackages="com.manuel.jordan.controller")
public class RequestMappingsViewConfig {

	@Bean
	public String requestMappingsView(){
		throw new RuntimeException("This @Configuration with @Bean NEVER must be created in runtime");
	}

}
```

Where `@RequestMappingsViewProfile` applies to the `request-mappings-view` profile

Thus the following is mandatory:

* `web-29-controller`, right click -> `Properties` 

and then

**First Tab**
![01](https://user-images.githubusercontent.com/2552795/34005286-7582f354-e0c8-11e7-9bd5-68f621e7c44b.png)

**Third Tab**
![02](https://user-images.githubusercontent.com/2552795/34005287-75a280d4-e0c8-11e7-9662-00cd58abbd45.png)

**Second Tab**
![03](https://user-images.githubusercontent.com/2552795/34005288-75c48fd0-e0c8-11e7-9e61-03c577b037ed.png)
![04](https://user-images.githubusercontent.com/2552795/34005289-75e2defe-e0c8-11e7-9f1a-b99b39241dd0.png)

**RequestMapping View**
![05](https://user-images.githubusercontent.com/2552795/34005290-7601d11a-e0c8-11e7-82b4-5fb19e9473f8.png)

Why according with the previous figure all the `Resource URL` have the `/` value?
For the selected item I expect see `/locale/changelocale`, it according the following code

```
@Controller
@RequestMapping(path="/locale")
public class LocaleController {

	...

	/**
	 * Inform the new current locale established by the user.
	 *
	 * @param model for internal view purposes
	 * @return view location
	 */
	@GetMapping(path="/changelocale")
	public String changeLocale(Model model){
         ...	
        }

}
```

Not sure what is wrong or missing

**Comment by martinlippert** [comment]
Hey @manueljordan can you attach a sample project?

**Comment by manueljordan** [comment]
Hello Martin

Sorry by the huge delay, the project sample is here: [sts-rmw](https://github.com/manueljordan/sts-rmw).

I share now the following experience in `STS` version `3.9.2` _and_ `Spring Framework` version `5.0.4`, below the situation:

Root and child project with `Spring Nature`
![01](https://user-images.githubusercontent.com/2552795/37867132-249af7e6-2f62-11e8-95e5-b47eaa73879a.png)

Scanned the `@Configuration` classes
![02](https://user-images.githubusercontent.com/2552795/37867133-24b9d274-2f62-11e8-9a5c-7ecebde2d2e3.png)
![03](https://user-images.githubusercontent.com/2552795/37867134-24d651ce-2f62-11e8-8711-11a2671abe7b.png)
![04](https://user-images.githubusercontent.com/2552795/37867135-24f28db2-2f62-11e8-89ca-2d500866d454.png)

For the `sts-rmw-spring` project opening the `@RequestMappings` view

![05](https://user-images.githubusercontent.com/2552795/37867136-2511ebe4-2f62-11e8-91bf-2a7c9638fa5c.png)

Nothing

![06](https://user-images.githubusercontent.com/2552795/37867137-252fb930-2f62-11e8-86b1-afb9cee48060.png)
![07](https://user-images.githubusercontent.com/2552795/37867138-254c9f96-2f62-11e8-8ae5-b3c4eaa5a499.png)

**Solution** `Clean` and `Build Project`

Thus appears the following

![08](https://user-images.githubusercontent.com/2552795/37867139-256addf8-2f62-11e8-8ff1-fe006bcdcb1f.png)

**Problem**: all the `Resource URL` have `/`

**Comment by martinlippert** [comment]
Thanks a lot for the detailed bug report and the sample project, that helped a lot to reproduce and fix this issue. Very much appreciated. Debugged this today and fixed the issue. Should be up tomorrow in the nightly distribution builds as well as published as part of STS 3.9.5.

**Comment by manueljordan** [comment]
Thanks to you for your valuable, quick and polite support ... I enjoy work in this IDE
